errbot/core_plugins/chatRoom.py

```diff
@@ -25,6 +25,8 @@ def callback_connect(self):
             self.connected = True
             for room in self.bot_config.CHATROOM_PRESENCE:
                 self.log.debug('Try to join room %s' % repr(room))
+                if 'password' not in self.bot_config.BOT_IDENTITY.keys():
+                    self.bot_config.BOT_IDENTITY['password']=None
```

Comment:
This change most definitely doesn't belong here. The chatroom plugin is generic to all backends so it should contain as little (and preferably no) backend-specific code at all.

It looks like maybe you were trying to account for a user not specifying a `password` in their `BOT_IDENTITY` on IRC? If that is the case, it should be handled in the `__init__` of the IRC backend itself.